Client/Frontend/Library/HistoryPanel/HistoryPanel.swift

```diff
@@ -0,0 +1,651 @@
+                isEnabled = hasRecentlyClosed
+            default: break
+            }
+        }
+        
+        // Set interaction behavior and style
+        if isEnabled {
+            cell.titleLabel.alpha = 1.0
+            cell.leftImageView.alpha = 1.0
+            cell.selectionStyle = .default
+            cell.isUserInteractionEnabled = true
+        } else {
+            cell.titleLabel.alpha = 0.5
+            cell.leftImageView.alpha = 0.5
+            cell.selectionStyle = .none
+            cell.isUserInteractionEnabled = false
+        }
+        
+    }
+
+    func reloadData() {
+        
+        self.tableView.reloadData()
+        self.updateEmptyPanelState()
+        
+        if let cell = self.clearHistoryCell {
+//            setTappableStateAndStyle(enabled: !self.viewModel.groupedSites.isEmpty, forCell: cell)
+        }
+        
+    }
+    
+    // MARK: - Datasource helpers
+    
+    func removeHistoryItem(at indexPath: IndexPath) {
+        guard let siteItem = diffableDatasource?.itemIdentifier(for: indexPath) as? Site else { return }
+        
+        viewModel.removeSiteItem(site: siteItem, at: indexPath.section)
+        
+        updateEmptyPanelState()
+        
+        if let historyActionableCell = clearHistoryCell {
+            setTappableStateAndStyle(with: HistoryActionablesModel.activeActionables.first, on: historyActionableCell)
+        }
+        
+        applySnapshot(animatingDifferences: true)
+    }
+    
+    func siteAt(indexPath: IndexPath) -> Site? {
+        guard let siteItem = diffableDatasource?.itemIdentifier(for: indexPath) as? Site else { return nil }
+        
+        return siteItem
+    }
+
+    private func showClearRecentHistory() {
+        clearHistoryHelper.showClearRecentHistory(onViewController: self, didComplete: { [weak self] date in
+            if let date = date {
+                self?.viewModel.removeVisibleSectionFor(date: date)
+            } else {
+                // The only time there's no date, is when we are deleting everything.
+                self?.viewModel.visibleSections = []
+            }
+            
+            self?.viewModel.reloadData()
+            self?.applySnapshot(animatingDifferences: true)
+            
+            if let cell = self?.clearHistoryCell {
+                self?.setTappableStateAndStyle(with: HistoryActionablesModel.activeActionables.first(where: { $0.itemIdentity == .clearHistory }), on: cell)
+            }
+        })
+    }
+    
+    func siteForIndexPath(_ indexPath: IndexPath) -> Site? {
+        // First section is reserved for fixed history actionables.
+        guard indexPath.section > HistoryPanelSections.additionalHistoryActions.rawValue else {
+            return nil
+        }
+
+        let sitesInSection = viewModel.groupedSites.itemsForSection(indexPath.section - 1)
+        return sitesInSection[safe: indexPath.row]
+    }
+
+    func handleNotification(_ notification: Notification) {
```

Comment:
We should have this in a separate extension with the `Notifiable` protocol for easy navigation and better separation.